src/app/ReadClient.cpp

```diff
@@ -287,6 +248,22 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
         mpExchangeCtx = nullptr;
         SuccessOrExit(err);
     }
+    else if (aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
+    {
+        Protocols::InteractionModel::Status status;
+        VerifyOrExit(apExchangeContext == mpExchangeCtx, err = CHIP_ERROR_INCORRECT_STATE);
+        err           = StatusResponse::ProcessStatusResponse(apExchangeContext, std::move(aPayload), status);
+        mpExchangeCtx = nullptr;
```

Comment:
Why?  The `exit:` block belowe will handle that, since we always end up in an error state here, right?

Reply:
agree, removed, thanks